### DISCUSSION TITLE
hero plugin - solving somehow real colors for color/class picker

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#4646] [Ideas]
## Motivation
Improvement, color/class picker has the wrong colors, it is really confusing
It shows this:
![image](https://user-images.githubusercontent.com/5429548/148037105-2dd02ee3-37fd-4f67-bd3d-5e26bb042570.png)

But frontend colors are different:
![image](https://user-images.githubusercontent.com/5429548/148037197-c2d34e87-7d4f-4fb6-b1d5-988ecfa453b1.png)

## Proposed Solution
There are too many ways, no idea what is the best.

There are 2 areas:
1. correct classes
Hero plugin already using its own hero classes (for example for carousel)
it could be bootstrap alternatives like hero-bg-primary etc 

2. override core classes with theme
for example: 
 - load frontend theme file hero.css into hero plugin, if this file exists  (not full frontend css)
 
Or 
 - add scope for hero plugin (or admin area)
 
 (I think that admin scope for frontend theme is not used and without separating bg classes it would change bootstrap3 theme too)

**Comment by NECPWAweb** [comment]
Classes are fine in 2.3.3 and Bootstrap 5 theme with Hero so hopefully resolved but...

What size should the images/videos be?

Trying to upload direct via Hero admin it keeps saying img to large but you can select the same images from media manager

**Reply by tgtje** [reply]
late, 
but anyway : too large usually means the overall image size.  it displays on 400 px x 400 px.

it will accept around 800 px images in width/height during upload, but template gives {SETIMAGE: w=400&h=400} 
so if image used had been 694x 463 some cropping will occur. So it seems possible to alter this setting, but not tested if it will disrupt the overall behaviour.

**Reply by NECPWAweb** [reply]
Thanks, that does help.

**Reply by tgtje** [reply]
YW, just side note it belongs to the image 'on top' (right side); not background as they get 'stretched' (eg from 615 px to 2560 or alike)

**Comment by Jimmi08** [comment]
> Classes are fine in 2.3.3 and Bootstrap 5 theme with Hero so hopefully resolved but...
 
Probably because both use default bootstrap 5 classes (the same colors) 
I can see from CSS that in the admin area (plugin configuration) is bootstrap 3 or admin CSS used, so those colors are different for the frontend theme.
